### PR TITLE
hdf5: fix dev output, remove refs to /build from compiler wrappers

### PR DESCRIPTION
**Pull request by markuskowa**
###### Description of changes
Since version 1.14.0 the compiler wrappers contain the `AM_CPPFLAGS` that HDF5 was built with. These flags however, point to an include path in `/build` (from the sandbox). For most packages, this has no further consequences. However, some cmake
scripts ingest this non-existing path and fail. 
This patch simply removes the reference to build from the compiler wrappers, which solves the problem.

(See also https://github.com/Nix-QChem/NixOS-QChem/issues/262).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by markuskowa** [comment]
@GrahamcOfBorg build openmolcas
@GrahamcOfBorg build qgis

File: pkgs/tools/misc/hdf5/default.nix [hunk 1]
```diff
@@ -91,6 +91,20 @@ stdenv.mkDerivation rec {
     moveToOutput 'bin/h5pcc' "''${!outputDev}"
   '';
 
+  postFixup = ''
+    # Remove reference to /build, which get introduced
```
Comment (sheepforce) [inline pkgs/tools/misc/hdf5/default.nix:95]: If you would move the comment out of the bash line the hash of the derivation would not change upon just changing the comment.

**Review by ck3d — APPROVED** [review]
LGTM

**Comment by symphorien** [comment]
this should probably target staging

**Comment by markuskowa** [comment]
@symphorien it acually is a direct fix of problem, rather than an update or an improvement. That's why I want to merge it into master.

**Comment by symphorien** [comment]
I was told on matrix that if the issue this fixes is only present on staging it should be targeting staging-next, and otherwise staging.

**Comment by markuskowa** [comment]
@symphorien My understanding was, that direct fixes for problems on master do not need to go through staging first.

**Pull request merged**